lisp/mastodon-notifications.el

```diff
@@ -0,0 +1,136 @@
+;;; mastodon-notifications.el --- Notification functions for mastodon.el -*- lexical-binding: t -*-
+
+;; Copyright (C) 2017 Johnson Denen
+;; Author: Johnson Denen <johnson.denen@gmail.com>
+;; Version: 0.7.2
+;; Homepage: https://github.com/jdenen/mastodon.el
+;; Package-Requires: ((emacs "24.4"))
+
+;; This file is not part of GNU Emacs.
+
+;; This file is part of mastodon.el.
+
+;; mastodon.el is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; mastodon.el is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with mastodon.el.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; mastodon-notification.el provides notification functions for Mastodon.
+
+;;; Code:
+
+(defvar mastodon-notifications--types-alist
+  '(("mention" . mastodon-notifications--mention)
+    ("follow" . mastodon-notifications--follow)
+    ("favourite" . mastodon-notifications--favourite)
+    ("reblog" . mastodon-notifications--reblog))
+  "Alist of notification types and their corresponding function.")
+
+(defvar mastodon-notifications--response-alist
+  '(("Mentioned" . "you")
+    ("Followed" . "you")
+    ("Favourited" . "your status")
+    ("Boosted" . "your status"))
+  "Alist of subjects for notification types.")
+
+(defun mastodon-notifications--byline-concat (message)
+  "Add byline for TOOT with MESSAGE."
+  (concat
+   " "
+   (propertize message 'face 'highlight)
+   " "
+   (cdr (assoc message mastodon-notifications--response-alist))))
+
+(defun mastodon-notifications--mention (note)
+  "Format for a `mention' NOTE."
+  (let ((status (mastodon-tl--field 'status note)))
+    (mastodon-tl--insert-status
+     note
+     (replace-regexp-in-string
```

Comment:
Might this be a good time to extract this regexp monster into a helper function somewhere?